doc/diagnostics/AddingTracepoints.md

```diff
@@ -0,0 +1,417 @@
+
+1) First determine the module you are about to add tracepoints to.
+
+The name that OpenJ9 trace uses to identify a module/component can be found in the module's trace definition (.tdf) file.  You should find a .tdf file in your module's root directory.  For example, the .tdf file for the j9vm component is runtime/vm/j9vm.tdf. The first line in that file: `Executable=j9vm` specifies the name that OpenJ9 trace (and -Xtrace options) use to identify tracepoints in that component. The .tdf file contains the specifications for all tracepoints in the j9vm component, including their format strings. Files `ut_j9vm.h` and `ut_j9vm.c`, are generated at build time from the contents of the .tdf.
+
+2) Step two is to add the tracepoint to your .tdf file. All new tracepoints must be added to the end of the .tdf file. This ensures that tracepoints preserve their tracepoint numbers between releases, see rules above.
+
+For example:
+
+`TraceEvent=Trc_VM_getMethodOrFieldID_dereferencedClass Overhead=1 Level=3 Template="class=%p (%.*s)"`
+
+This breaks down as follows:
+
+`TraceEvent=Trc_VM_getMethodOrFieldID_dereferencedClass`  This specifies the type and the name of the tracepoint. `Trc_VM_getMethodOrFieldID_dereferencedClass` is the name of the generated macro to use in your code. The `TraceEvent=` denotes this is an event trace point. Other tracepoint types are:  `TraceEntry` `TraceExit` `TraceException` `TraceExit-Exception` `TraceAssert` `TraceDebug`.
+
+`Overhead=1`  Each tracepoint has an overhead to determine whether it is included in the generated object code at build time. This is useful for very hot tracepoints, where even the cost of the simple 'if enabled' test at runtime is too great. The threshold in the tracepoint pre-processor is currently set to 1, so if you set Overhead to a value greater than 1, the tracepoint will be no-oped out at build time. It can still be useful to have the tracepoint defined in the source code, but of course requires a custom re-build of the relevant library as and when actually needed for diagnostics.
+
+`Level=3`  Each tracepoint has a level, used to control the volume of trace production at runtime. Level 1 and 2 tracepoints are switched on by default, and are written to the cyclic memory buffers for use in the flight recorder GC history and snap trace dumps. Level 2 trace points are enabled after the VM exits the startup phase (or before that if any other trace settings are enabled). Level 2 should be used for trace points that should be on by default once an application is running but are too frequently hit during the startup phase of an application to be in level 1. For example class loading trace points. If a tracepoint in the default set causes or is likely to cause a performance problem even outside of startup, it is raised to level 3 or higher. Levels can be 0-9, and of those levels, 1 (on by default), 2 (on by default, after startup), 3, 4, 5 and 6 are commonly used to reflect increasing detail.
```

Comment:
Please clarify whether level 0 is enabled by default.

Reply:
I'm not sure, I didn't realize level 0 was supported, and I'm not sure how it would be used. This pre-existing doc implies that level 0 is not enabled by default. I'll see what I can figure out from the code.

Reply:
Added statement "Level 0 is not on by default."

Reply:
https://github.com/eclipse/openj9/blob/master/runtime/rastrace/trccomponent.c#L1380 suggests level 0 is enabled when level 1 (or higher) is enabled.

Reply:
I should have realized that, even the User Guide states level 0 is enabled.